### PR TITLE
Add MongoDB extension

### LINKED ISSUE #302: Cerate a Camel MongoDB extension
https://github.com/quarkusio/quarkus/tree/master/extensions/mongodb-client
https://github.com/apache/camel/tree/master/components/camel-mongodb

### PR DESCRIPTION
**Pull request by jamesnetherton**
fixes #302

### CODE REVIEW

File: extensions/mongodb/runtime/src/main/java/org/apache/camel/quarkus/component/mongodb/CamelMongoClientProducer.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.mongodb.client.MongoClient;
+
+@ApplicationScoped
+public class CamelMongoClientProducer {
+
+    @Produces
+    @Singleton
+    @Named("camelMongoClient")
+    public com.mongodb.MongoClient camelMongoClient(MongoClient delegate) {
```
Comment (ppalaga) [inline extensions/mongodb/runtime/src/main/java/org/apache/camel/quarkus/component/mongodb/CamelMongoClientProducer.java]: Can't we install the `com.mongodb.MongoClient` into the Camel registry using `MongoClientBuildItem` ? Avoiding CDI could bring some performance benefits.

Comment (jamesnetherton) [reply extensions/mongodb/runtime/src/main/java/org/apache/camel/quarkus/component/mongodb/CamelMongoClientProducer.java]: I'll see if it's possible.

Comment (jamesnetherton) [reply extensions/mongodb/runtime/src/main/java/org/apache/camel/quarkus/component/mongodb/CamelMongoClientProducer.java]: Latest changes remove the CDI bits in favour of `MongoClientBuildItem`.

**Comment by ppalaga** [comment]
ok to test

**Pull request merged**